int32 thruster0
int32 thruster1